esp-wifi/src/timer/riscv.rs

```diff
@@ -28,6 +28,11 @@ const TIMER_DELAY: fugit::HertzU32 = fugit::HertzU32::from_raw(crate::CONFIG.tic
 static ALARM0: Mutex<RefCell<Option<Alarm<Periodic, 0>>>> = Mutex::new(RefCell::new(None));
 
 pub fn setup_timer(systimer: TimeBase) {
+    // make sure the scheduling won't start before everything is setup
+    unsafe {
```

Comment:
Would it be safer to wrap this in a CS? I think that with the P4 around the corner we might run into issues assuming riscv is single core?

Reply:
I guess it's enough to just make sure nothing happens on this core 
... and P4 won't get esp-wifi 😆 But sure there will be multicore RV chips with radio in future I bet

Reply:
Completely forgot P4 has no radio :laughing: